pkg/build/builder/dockerutil.go

```diff
@@ -62,15 +81,27 @@ func pushImage(client DockerClient, name string, authConfig docker.AuthConfigura
 	if glog.V(5) {
 		opts.OutputStream = os.Stderr
 	}
-	var err error
+
 	var retriableError = false
 
+	eventWatch, err := watchForFailedPushEvent(eventClient, ref.Namespace, ref.Name)
```

Comment:
The events related to ImageStreams should be pretty low on frequency. I don't think this is an ideal solution but I consider it safe given the filtering used. If, for some reason, we fail to catch an event, the user will still see the original `requested access to the resource is denied` message.